Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -222,6 +237,50 @@ public class PushDeploymentController : ApiController
             repository.Commit(zipDeploymentInfo.Message, zipDeploymentInfo.Author, zipDeploymentInfo.AuthorEmail);
         }
 
+        private async Task LocalZipFetchAndDeploy(IRepository repository, DeploymentInfoBase deploymentInfo, ILogger logger, ITracer tracer)
+        {
+            var zipDeploymentInfo = (ZipDeploymentInfo)deploymentInfo;
+
+            // For this kind of deployment, RepositoryUrl is a local path to the temp zip.
+            var sourceZipFile = zipDeploymentInfo.RepositoryUrl;
+            // the name of the deployed file in D:\home\data\SitePackages\{name}.zip is the 
+            // timestamp in the format yyyMMddHHmmss. 
+            var fileName = $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}.zip";
+            var filePath = Path.Combine(_environment.SitePackagesPath, fileName);
+
+            FileSystemHelpers.EnsureDirectory(_environment.SitePackagesPath);
+
+            var info = FileSystemHelpers.FileInfoFromFileName(sourceZipFile);
+            var sizeInMb = (info.Length / (1024f * 1024f)).ToString("0.00", CultureInfo.InvariantCulture);
+
+            var message = String.Format(
+                CultureInfo.InvariantCulture,
+                "Deploying {0} to {1} with size of {2}",
+                info.FullName,
+                filePath,
+                sizeInMb);
+
+            logger.Log(message);
+
+            using (tracer.Step(message))
+            {
+                // If extractTargetDirectory already exists, rename it so we can delete it concurrently with
+                // the unzip (along with any other junk in the folder)
+                var cleanTask = Task.Run(() => DeleteFilesAndDirsExcept(sourceZipFile, null, tracer));
+                var extractTask = Task.Run(() =>
+                {
+                    FileSystemHelpers.CopyFile(sourceZipFile, filePath, overwrite: true);
```

Comment:
Ideally, we would save it directly there without it ever being in temp. But I guess that would make the logic more complex?

Interestingly, at runtime, this zip will end up getting copied back somewhere into d:\local.

Let's make sure that the `sourceZipFile` gets deleted here at the end of deployment.